clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackLoadBalancerCachingAgent.groovy

```diff
@@ -45,80 +53,163 @@ import static com.netflix.spinnaker.clouddriver.openstack.cache.Keys.Namespace.V
   }
 
   @Override
   CacheResult loadData(ProviderCache providerCache) {
     log.info("Describing items in ${agentType}")
 
-    CacheResultBuilder cacheResultBuilder = new CacheResultBuilder(startTime: Long.MAX_VALUE)
-
     List<LbPool> pools = clientProvider.getAllLoadBalancerPools(region)
 
-    pools.collect { pool ->
-      //health monitors get looked up
-      Set<HealthMonitor> healthMonitors = pool.healthMonitors?.collect { healthId ->
-        clientProvider.getHealthMonitor(region, healthId)
-      }?.toSet()
-
-      //vips cached
-      Map<String, Object> vipMap = providerCache.get(VIPS.ns, Keys.getVipKey(pool.vipId, accountName, region))?.attributes
-      OpenstackVip vip = vipMap ? objectMapper.convertValue(vipMap, OpenstackVip) : null
-
-      //ips cached
-      OpenstackFloatingIP ip = null
-      if (vip) {
-        Collection<String> portFilters = providerCache.filterIdentifiers(PORTS.ns, Keys.getPortKey('*', accountName, region))
-        Collection<CacheData> portsData = providerCache.getAll(PORTS.ns, portFilters, RelationshipCacheFilter.none())
-        CacheData portCacheData = portsData?.find { p -> p.attributes?.name == "vip-${vip.id}" }
-        Map<String, Object> portAttributes = portCacheData?.attributes
-        OpenstackPort port = objectMapper.convertValue(portAttributes, OpenstackPort)
-        if (port) {
-          Collection<String> ipFilters = providerCache.filterIdentifiers(FLOATING_IPS.ns, Keys.getFloatingIPKey('*', accountName, region))
-          Collection<CacheData> ipsData = providerCache.getAll(FLOATING_IPS.ns, ipFilters, RelationshipCacheFilter.none())
-          CacheData ipCacheData = ipsData.find { i -> i.attributes?.portId == port.id }
-          Map<String, Object> ipAttributes = ipCacheData?.attributes
-          ip = objectMapper.convertValue(ipAttributes, OpenstackFloatingIP)
+    List<String> loadBalancerKeys = pools.collect { Keys.getLoadBalancerKey(it.name, it.id, accountName, region) }
+
+    buildLoadDataCache(providerCache, loadBalancerKeys) { CacheResultBuilder cacheResultBuilder ->
+      buildCacheResult(providerCache, pools, cacheResultBuilder)
+    }
+  }
+
+  CacheResult buildCacheResult(ProviderCache providerCache, List<LbPool> pools, CacheResultBuilder cacheResultBuilder) {
+    pools?.collect { pool ->
+      String loadBalancerKey = Keys.getLoadBalancerKey(pool.name, pool.id, accountName, region)
+
+      if (shouldUseOnDemandData(cacheResultBuilder, loadBalancerKey)) {
+        moveOnDemandDataToNamespace(objectMapper, typeReference, cacheResultBuilder, loadBalancerKey)
+      } else {
+        //health monitors get looked up
+        Set<HealthMonitor> healthMonitors = pool.healthMonitors?.collect { healthId ->
+          clientProvider.getHealthMonitor(region, healthId)
+        }?.toSet()
+
+        //vips cached
+        Map<String, Object> vipMap = providerCache.get(VIPS.ns, Keys.getVipKey(pool.vipId, accountName, region))?.attributes
```

Comment:
Thanks for the heads up ... We are in the process of rewriting the majority of load balancer functionality to support lbaasv2 and remove lbaasv1.  I will add this comment to our caching ticket.